workflows/rnaseq/rnaseq-standard-fastq.wdl

```diff
@@ -0,0 +1,122 @@
+## # RNA-Seq Standard from FastQ
+##
+## This WDL workflow runs the STAR RNA-seq alignment workflow for St. Jude Cloud from fastq input.
+## The workflow takes paired input fastq files. 
+## The read pairs are then passed through STAR alignment to generate a BAM file. The BAM is run
+## through several QC steps. Quantification is done using htseq-count. 
+##
+## ## LICENSING
+##
+## #### MIT License
+##
+## Copyright 2019 St. Jude Children's Research Hospital
+##
+## Permission is hereby granted, free of charge, to any person obtaining a copy of this
+## software and associated documentation files (the "Software"), to deal in the Software
+## without restriction, including without limitation the rights to use, copy, modify, merge,
+## publish, distribute, sublicense, and/or sell copies of the Software, and to permit persons
+## to whom the Software is furnished to do so, subject to the following conditions:
+##
+## The above copyright notice and this permission notice shall be included in all copies or
+## substantial portions of the Software.
+##
+## THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING
+## BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+## NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM,
+## DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+## OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+
+version 1.0
+
+import "https://raw.githubusercontent.com/stjudecloud/workflows/master/tools/fq.wdl"
+import "https://raw.githubusercontent.com/stjudecloud/workflows/master/tools/star.wdl"
+import "https://raw.githubusercontent.com/stjudecloud/workflows/master/tools/picard.wdl"
+import "https://raw.githubusercontent.com/stjudecloud/workflows/master/tools/ngsderive.wdl"
+import "https://raw.githubusercontent.com/stjudecloud/workflows/master/tools/htseq.wdl"
+import "https://raw.githubusercontent.com/stjudecloud/workflows/master/tools/samtools.wdl"
+import "https://raw.githubusercontent.com/stjudecloud/workflows/master/tools/util.wdl"
+import "https://raw.githubusercontent.com/stjudecloud/workflows/master/tools/deeptools.wdl"
+
+workflow rnaseq_standard_fastq {
+    input {
+        File gencode_gtf
+        Array[File] read_one_fastqs
+        Array[File] read_two_fastqs
```

Comment:
Sorry if we went over this before, can we not support SE in RNA-Seq V2? We've enable it for STAR, can it work with the other tools? The header of this file would need to updated if we enable it, FYI

Reply:
ngsderive strandedness requires paired-end data, though we could conceivably update that in the future. htseq-count works with both SE and PE data. So does deeptools bamcoverage. fqlint would need to be skipped.

Reply:
Ok I agree we can leave it required for now. We can make those changes when the use-case comes up.